src/ConfigurationStore.php

```diff
@@ -18,30 +19,34 @@ public function __construct(CacheInterface $cache)
         $this->cache = $cache;
     }
 
-    /**
-     * @param string $key
-     *
-     * @return array
-     *
-     * @throws InvalidArgumentException
-     */
-    public function getConfiguration(string $key): array
+    public function setFlag(Flag $flag): void
     {
-        $value = $this->cache->get($key);
-        return $value ? json_decode($value, true) : [];
+        try {
+            $this->cache->set($flag->key, serialize($flag));
+        } catch (InvalidArgumentException $e) {
+            $key = $flag->key;
+            syslog(LOG_WARNING, "[EPPO SDK] Invalid flag key ${key}: " . $e->getMessage());
```

Comment:
How can a flag key be invalid? If it's empty?

Reply:
empty string (though implementations appear to support the empty string) and the following characters will cause the throw
{}()/\@:
src: https://www.php-fig.org/psr/psr-16/